cmd/config.go

```diff
@@ -309,6 +301,85 @@ type CAADistributedResolverConfig struct {
 	Proxies     []string
 }
 
+type TemporalLogDescription struct {
+	URI         string
+	Key         string
+	WindowStart string
+	start       time.Time
+	WindowEnd   string
+	end         time.Time
+}
+
+type TemporalSet struct {
+	Name   string
+	Shards []TemporalLogDescription
+}
+
+// Setup initializes the TemporalSet by parsing the start and end dates
+// and verifying the intervals
+func (ts *TemporalSet) Setup() error {
```

Comment:
I think `Setup()` should probably throw an error `if len(ts.Shards) == 0` since that's definitely some kind of misconfiguration.

Reply:
Should `Setup()` be checking the individual shards against one another to ensure they're consistent? I think presently you could create overlapping shards and based on `pick()` only the first would ever be used.

Reply:
Overlapping shards are valid. It's preference in that case which shard you pick, I just went for the earliest.